UsefulProteomicsDatabases/UnimodLoader.cs

```diff
@@ -54,24 +63,49 @@ internal static IEnumerable<ModificationWithLocation> ReadMods(string unimodLoca
                 {
                     var tg = nice.site;
                     if (tg.Length > 1)
-                        tg = "X";
+                        tg = "X"; //I think that we should allow motifs here using the trygetmotif
                     ModificationMotif.TryGetMotif(tg, out ModificationMotif motif);
-                    var pos = nice.position;
-                    IDictionary<string, IList<string>> dblinks = new Dictionary<string, IList<string>>
+
+                    if (positionConversion.TryGetValue(nice.position.ToString(), out string pos))
+                    {
+                        //do nothing, the new string value should be there
+                    }
+                    else
+                    {
+                        pos = null;
+                    }
+
+                    Dictionary<string, IList<string>> dblinks = new Dictionary<string, IList<string>>
                     {
                         { "Unimod",  new List<string>{ac.ToString() } },
                     };
 
                     if (nice.NeutralLoss == null)
-                        yield return new ModificationWithMassAndCf(id + " on " + motif + " at " + positionDict[pos], "Unimod", motif, positionDict[pos], cf, linksToOtherDbs: dblinks);
+                        yield return new ModificationGeneral(_id: id, _modificationType: "Unimod", _target: motif, _locationRestriction: pos, _chemicalFormula: cf, _databaseReference: dblinks);
                     else
                     {
-                        List<double> neutralLosses = new List<double>();
+                        Dictionary<MassSpectrometry.DissociationType, List<double>> neutralLosses = null;
                         foreach (var nl in nice.NeutralLoss)
                         {
                             ChemicalFormula cfnl = new ChemicalFormula();
                             if (nl.mono_mass == 0)
-                                neutralLosses.Add(0);
+                            {
+                                if (neutralLosses == null)
+                                {
+                                    neutralLosses = new Dictionary<MassSpectrometry.DissociationType, List<double>>();
+                                    neutralLosses.Add(MassSpectrometry.DissociationType.AnyActivationType, new List<double> { 0 });
+                                }
+                                else
+                                {
```

Comment:
unimod doesn't have neutral losses but we need to work on this section if it ever does